catalog/core/catalog-core-dynamic/src/main/java/ddf/catalog/data/dynamic/impl/DynamicMetacard.java

```diff
@@ -0,0 +1,644 @@
+        }
+        return descriptors;
+    }
+
+    /**
+     * Returns the {@link AttributeDescriptor} for the specified attribute.
+     *
+     * @param s name of the attribute whose descriptor is to be returned
+     * @return the {@link AttributeDescriptor} for the named attribute
+     */
+    @Override
+    public AttributeDescriptor getAttributeDescriptor(String s) {
+        return null;
+    }
+
+    /**
+     * Adds another value to the given attribute. If the attribute is not a multi-valued
+     * attribute, it replaces the current value with the provided value.
+     *
+     * @param name  name of the attribute to set
+     * @param value the value to be set
+     */
+    public void addAttribute(String name, Object value) {
+        DynaProperty dynaProperty = attributesBean.getDynaClass()
+                .getDynaProperty(name);
+        if (value instanceof URI) {
+            value = ((URI) value).toString();
+        }
+
+        try {
+            if (dynaProperty == null) {
+                attributesBean.set(name, 0, value);
+            } else {
+                if (dynaProperty.isIndexed()) {
+                    attributesBean.set(name, attributesBean.size(name), value);
+                } else {
+                    LOGGER.warn(
+                            "Can't add another value to a simple attribute {} - replacing value.",
+                            name);
+                    attributesBean.set(name, value);
+                }
+            }
+        } catch (ConversionException e) {
+            LOGGER.warn(
+                    "Unable to to convert provided value of class {} to attribute {} value of class {} - no action taken.",
+                    value.getClass()
+                            .getSimpleName(),
+                    name,
+                    dynaProperty.getContentType()
+                            .getSimpleName());
+        } catch (IllegalArgumentException e) {
+            LOGGER.warn("Attribute {} has different than expected cardinality - no action taken",
+                    name);
+        } catch (IndexOutOfBoundsException e) {
+            LOGGER.warn("Trying to set attribute {} at index {} which is out of range.",
+                    name,
+                    attributesBean.size(name));
+        } catch (NullPointerException e) {
+            LOGGER.warn(
+                    "Trying to set primitive type for attribute {} to a null value - no action taken",
+                    name);
+        }
+    }
+
+    /**
+     * Sets the value of the given attribute. If the attribute is a multi-valued
+     * attribute, it adds the given value to the current collection.
+     *
+     * @param name  name of the attribute to set
+     * @param value the value to be set
+     */
+    public void setAttribute(String name, Object value) {
+        DynaProperty dynaProperty = attributesBean.getDynaClass()
+                .getDynaProperty(name);
```

Comment:
It looks like this could throw an IllegalArguemenException if the property isn't found.  Is this OK or should this be caught and logged here?